packages/client/wallets/smart-wallet/src/blockchain/wallets/EVMSmartWallet.ts

```diff
@@ -41,79 +38,47 @@ export class EVMSmartWallet {
     }
 
     /**
+     * Transfers tokens from the smart wallet to a specified address.
+     * @param toAddress The recipient's address.
+     * @param config The transfer configuration, including token details and amount.
      * @returns The transaction hash.
+     * @throws {SmartWalletError} If there's a chain mismatch between this wallet and the input configuration.
+     * @throws {SendTransactionError} If the transaction fails to send. Contains the error thrown by the viem client.
+     * @throws {SendTransactionExecutionRevertedError} A subclass of SendTransactionError if the transaction fails due to a contract execution error.
      */
-    public async transferToken(toAddress: string, config: TransferType): Promise<string> {
-        return this.logger.logPerformance(
-            "TRANSFER",
-            async () => {
-                if (this.chain !== config.token.chain) {
-                    throw new Error(
-                        `Chain mismatch: Expected ${config.token.chain}, but got ${this.chain}. Ensure you are interacting with the correct blockchain.`
-                    );
-                }
-
-                if (!isAddress(toAddress)) {
-                    throw new Error(`Invalid recipient address: '${toAddress}' is not a valid EVM address.`);
-                }
+    public async transferToken(toAddress: Address, config: TransferType): Promise<string> {
+        if (this.chain !== config.token.chain) {
+            throw new SmartWalletError(
+                `Chain mismatch: Expected ${config.token.chain}, but got ${this.chain}. Ensure you are interacting with the correct blockchain.`
+            );
+        }
 
-                if (!isAddress(config.token.contractAddress)) {
-                    throw new Error(
-                        `Invalid contract address: '${config.token.contractAddress}' is not a valid EVM address.`
-                    );
-                }
-
-                const tx = transferParams({
-                    contract: config.token.contractAddress,
-                    to: toAddress,
-                    from: this.accountClient.account,
-                    config,
-                });
-
-                try {
-                    const client = this.accountClient.extend(publicActions);
-                    const { request } = await client.simulateContract(tx);
-                    const hash = await client.writeContract(request);
-                    this.logger.log(`[TRANSFER] - Transaction hash from transfer: ${hash}`);
-
-                    return hash;
-                } catch (error) {
-                    this.logger.error("[TRANSFER] - ERROR_TRANSFERRING_TOKEN", {
-                        error: error instanceof Error ? error.message : JSON.stringify(error),
-                        tokenId: tx.tokenId,
-                        contractAddress: config.token.contractAddress,
-                        chain: config.token.chain,
-                    });
```

Comment:
Because of the decoration on our `viem` client here, we already log errors here.